RUFAS/biophysical/animal/animal_growth/animal_growth.py

```diff
@@ -1,2 +1,412 @@
+from RUFAS.biophysical.animal.animal_properties.general_properties import GeneralProperties
+from RUFAS.biophysical.animal.animal_properties.reproduction_properties import ReproductionProperties
+from RUFAS.biophysical.animal.animal_properties.animal_growth_properties import AnimalGrowthProperties
+
+
 class AnimalGrowth:
-    pass
+    """
+    Handles updating the body weight growth related animal attributes.
+
+    Attributes
+    ----------
+    wean_day: int
+        Class constant that indicates the user-defined wean day for calves.
+    target_heifer_pregnant_day: int
+        Class constant that indicates the user-defined target pregnant day for heifers.
+    """
+
+    wean_day: int
+    target_heifer_pregnant_day: int
+
+    @classmethod
+    def initialize_class_variables(cls) -> None:
+        """
+        This function retrieves the user input data from the InputManager and initializes the class constants.
+        """
+        im = InputManager()
+        animal_config: dict[str, dict[str, Any]] = im.get_data("animal.animal_config.farm_level")
+        cls.wean_day = animal_config["calf"]["wean_day"]
+        cls.target_heifer_pregnant_day = animal_config["bodyweight"]["target_heifer_preg_day"]
+
+    @staticmethod
+    def daily_routines(
+        general_properties: GeneralProperties,
+        animal_growth_properties: AnimalGrowthProperties,
+        reproduction_properties: ReproductionProperties,
+        time: Time,
+    ) -> tuple[AnimalGrowthProperties, ReproductionProperties, GeneralProperties]:
+        """
+        Handles an animal's daily growth updates.
+
+        Parameters
+        ----------
+        general_properties: GeneralProperties
+            Animal properties that are general or are used to determine many animal outcomes.
+        animal_growth_properties: AnimalGrowthProperties
+            Animal properties that are related to body weight growth.
+        reproduction_properties: ReproductionProperties
+            Animal properties that are related to animal reproduction.
+        general_properties : GeneralProperties
+        time : Time
+            Time instance containing the current time of the simulation.
+
+        Returns
+        -------
+        tuple[AnimalGrowthProperties, ReproductionProperties, GeneralProperties]
+            The updated animal growth properties, reproduction properties, and the general properties of the animal
+            after the growth-related routines for the current day.
+        """
+        if general_properties.animal_type == AnimalType.CALF:
+            animal_growth_properties.daily_growth = AnimalGrowth.calculate_calf_body_weight_change(general_properties)
+            general_properties.body_weight += animal_growth_properties.daily_growth
+
+        elif general_properties.animal_type == AnimalType.HEIFER_I or (
+            general_properties.animal_type == AnimalType.HEIFER_II and not general_properties.is_pregnant
```

Comment:
If possible, I think for each of these elif statements it would be helpful to extract the animal groupings into boolean vars that explain why they're grouped like that for this purpose.

e.g. (just made up this grouping so not please don't take it as what should actually be used)

```suggestion
        is_non_pregnant_heifer = general_properties.animal_type == AnimalType.HEIFER_I or (
            general_properties.animal_type == AnimalType.HEIFER_II and not general_properties.is_pregnant
        elif is_non_pregnant_heifer:

```

Reply:
Good point!